R/smoking-status.R

```diff
@@ -0,0 +1,100 @@
+#' @title Simple smoking status
+#'
+#' @description This function creates a derived smoking variable (smoke_simple)  
+#'  with four categories: 
+#'  
+#' \itemize{
+#'   \item non-smoker (never smoked)
+#'   \item current smoker (daily and occasional?)
+#'   \item former daily smoker quit ≤5 years or former occasional smoker 
+#'   \item former daily smoker quit >5 years
+#'  }
+#'
+#' @param SMKDSTY derived variable that classifies an individual's smoking
+#'  status.
+#'
+#' @param SMK_09A_B number of years since quitting smoking. Variable asked to
+#'  former daily smokers who quit <3 years ago.
+#'
+#' @param SMKG09C number of years since quitting smoking. Variable asked to
+#'  former daily smokers who quit >=3 years ago.
+#'
+#' @example
+#' # Using the 'derive_smoke_simple' function to create the derived smoking   
+#' # variable across CCHS cycles.
+#' # derive_smoke_simple() is specified in the variable_details.csv
+#'
+#' # To create a harmonized smoke_simple variable across CCHS cycles, use 
+#' # rec_with_table() for each CCHS cycle and specify derive_smoke_simple and 
+#' # the required base variables.
+#' # Using bind_rows(), you can combine smoke_simple across cycles
+#'
+#' library(cchsflow)
+#'
+#' smoke_simple2009_2010 <- rec_with_table(
+#'   cchs2009_2010_p, c(
+#'     "SMKDSTY", "SMK_09A_B", "SMKG09C", "derive_smoke_simple"
+#'   )
+#' )
+#'
+#' head(smoke_simple2009_2010)
+#'
+#' smoke_simple2011_2012 <- rec_with_table(
+#'   cchs2011_2012_p,c(
+#'     "SMKDSTY", "SMK_09A_B", "SMKG09C", "derive_smoke_simple"
+#'   )
+#' )
+#'
+#' tail(smoke_simple2011_2012)
+#'
+#' combined_smoke_simple <- suppressWarnings(bind_rows(smoke_simple2009_2010,
+#'  smoke_simple2011_2012))
+#'
+#' head(combined_smoke_simple)
+#' tail(combined_smoke_simple)
+#' @export
+derive_smoke_simple <-
```

Comment:
Keeping with the function naming convention I think we should call it `smoke_simple_fun`.

Reply:
@yulric I was striving to follow the style guide:  "...function names should be verbs. Strive for names that are concise and meaningful.." I'm fine to go with your suggestion as long as we are happy with veering from the style guide.

Reply:
Yes, I believe that's fine in this case.

Reply:
Done